File: Working with Objects/WorkingWithObjects/WorkingWithObjects/XYZPerson.m
//
//  XYZPerson.m
//  WorkingWithObjects
//
//  Created by 김기현 on 2021/02/15.
//

#import <Foundation/Foundation.h>
#import "XYZPerson.h"

@implementation XYZPerson

- (void) sayHello: (NSString *) name {
    [self saySomething: name];
} // sayHello

- (void) saySomething: (NSString *) greet {
    NSLog(@"%@", greet);
} // saySomething

- (int) maginNumber {
    return 42;
} // magicNumber

- (NSString *) magicString: (NSString *) input {
    return [input uppercaseString];
} // magicString

@end // XYZPerson

File: XYZPerson Swift/XYZPersonSwift/XYZPersonSwift/XYZPerson.swift
//
//  XYZPerson.swift
//  XYZPersonSwift
//
//  Created by 김기현 on 2021/02/17.
//

import Foundation

class XYZPerson {
    var firstName: String
    var lastName: String
    
    init(aFirstName: String, aLastName: String) {
        self.firstName = aFirstName
        self.lastName = aLastName
    }
    
    init() {
        self.firstName = "John"
        self.lastName = "Doe"
    }
    
    func sayHello(name: String) {
        self.saySomething(greet: name)
    }
    
    func saySomething(greet: String) {
        print("Hello, \(greet)")
    }
    
    func magicNumber() -> Int {
        return 42
    }
    
    func magicString(input: String) -> String {
        return input.uppercased()
    }
    
    func getFullName() -> String {
        return self.firstName + " " + self.lastName
    }
}
